src/legacy/ui/public/chrome/directives/header_global_nav/components/header.tsx

```diff
@@ -238,85 +232,39 @@ class HeaderUI extends Component<Props, State> {
-                  aria-label="Recently viewed items"
-                  onClick={() => this.expandFlyout()}
-                  isDisabled={recentlyAccessed.length > 0 ? false : true}
-                  extraAction={{
-                    color: 'subdued',
-                    iconType: 'arrowRight',
-                    iconSize: 's',
-                    'aria-label': 'Expand to view recent apps and objects',
-                    onClick: () => this.expandFlyout(),
-                    alwaysShow: true,
-                  }}
-                />
-              </EuiListGroup>
-              <EuiHorizontalRule margin="none" />
-              <EuiListGroup data-test-subj="appsMenu">
-                {navLinks.map(navLink =>
-                  navLink.hidden ? null : (
-                    <EuiListGroupItem
-                      key={navLink.id}
-                      label={navLink.title}
-                      href={navLink.href}
-                      iconType={navLink.euiIconType}
-                      size="s"
-                      style={{ color: 'inherit' }}
-                      aria-label={navLink.title}
-                      isActive={navLink.active}
-                      data-test-subj="appLink"
-                    />
-                  )
-                )}
-              </EuiListGroup>
-            </EuiNavDrawerMenu>
-            <EuiNavDrawerFlyout
-              id="navDrawerFlyout"
-              title="Recent items"
-              isCollapsed={this.state.flyoutIsCollapsed}
-              listItems={recentlyAccessed.map(item => ({
-                label: item.label,
-                href: item.href,
-                iconType: item.euiIconType,
-                size: 's',
-                style: { color: 'inherit' },
-                'aria-label': item.label,
-              }))}
-              onMouseLeave={this.collapseFlyout}
-              wrapText={true}
+        <EuiNavDrawer ref={this.setNavDrawerRef} data-test-subj={classNames('navDrawer')}>
+          <EuiListGroup>
+            <EuiListGroupItem
+              label="Recently viewed"
+              iconType="clock"
+              size="s"
+              style={{ color: 'inherit' }}
+              aria-label="Recently viewed items"
+              isDisabled={recentlyAccessed.length > 0 ? false : true}
+              flyoutMenu={{
+                title: 'Recent items',
+                listItems: recentlyAccessed.map(item => ({
+                  label: item.label,
+                  href: item.href,
+                  iconType: item.euiIconType,
+                  size: 's',
+                  style: { color: 'inherit' },
+                  'aria-label': item.label,
+                })),
+              }}
+              extraAction={{
+                color: 'subdued',
+                iconType: 'arrowRight',
+                iconSize: 's',
+                'aria-label': 'Expand to view recent apps and objects',
+                onClick: this.expandFlyout,
```

Comment:
Do you think the extra action is necessary on these? I feel like they're only causing more problems than they're worth. Also they're the same onClick action as the regular button, so it's not actually providing a secondary or separate functionality.

Reply:
It is definitely causing more pain than it's worth and it doesn't seem like a good approach to set this automatically on the EUI side, either. Not only is it redundant, but there could be a future use case where you'd want to use this for another action (pin/re-order) which would be overwritten if we used this detail arrow/expand function. I'll remove it.